include/envoy/network/filter.h

```diff
@@ -331,6 +331,11 @@ class FilterChain {
    * const std::vector<FilterFactoryCb>& a list of filters to be used by the new connection.
    */
   virtual const std::vector<FilterFactoryCb>& networkFilterFactories() const PURE;
+
+  /**
+   * @return uint64_t the tag the filter chain should use for connection handler tracking.
+   */
+  virtual uint64_t filterChainTag() const PURE;
```

Comment:
I don't think I understand why we need a tag here, especially as part of the public interface. Can you explain why this is needed? At a high level it seems like we need a shared_ptr to any shared stat which is snapped for each connection, and possible weak pointers to avoid circular dependency, though I don't know the specifics without spending more time with the code myself.

If we remove tag, I think we can also get rid of `FilterChainFactoryContext` from the public interface and have this all be part of the implementation itself.

Reply:
I am using Network::FilterChain raw pointer to replace the usage of Network::FilterChain::tag()

Choose raw_ptr because we have only raw pointer [here](https://github.com/envoyproxy/envoy/blob/bfc7602ec77e68429d9bad9af80e8cb648c6dd77/source/server/filter_chain_manager_impl.h#L34-L35)

Reply:
I prefer not to remove `FilterChainFactoryContext` from the public interface for now.

FilterChainFactoryContext and FactoryContext has the exact function signatures, however, another implementation of FactoryContext is ListenerContext. In production we must adopt `FilterChainFactoryContext` because of the lifetime requirement.